modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php

```diff
@@ -119,6 +119,14 @@ protected function addRenderableLocation(array &$build, ContentEntityInterface $
       return;
     }
 
+    if (!$venue->access('view', \Drupal::currentUser())) {
+      return;
+    }
+
+    if (!$venue->isPublished()) {
+      return;
+    }
+
```

Comment:
It's probably not needed. There is permission to view published and and view unpublished entities.
Check build/modules/contrib/oe_content/modules/oe_content_entity/src/PermissionCallbacksBase.php